x/programs/rust/wasmlanche_sdk/src/simulator.rs

```diff
@@ -0,0 +1,363 @@
+//! A client and types for the VM simulator.
+
+use std::{
+    error::Error,
+    io::Write,
+    process::{Command, Stdio},
+};
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum Endpoint {
+    /// Perform an operation against the key api.
+    Key,
+    /// Make a read-only call to a program function and return the result.
+    View,
+    /// Create a transaction from a possible state changing program function
+    /// call. A program's function can internally optionally call other
+    /// functions including program to program.
+    Execute,
```

Comment:
If this is solely related to transactions maybe we can call it `Transactions` or `Transact`? It feels a bit vague.

Reply:
But I do want separation for non transaction events

Reply:
I think we will leave this as it does logically flow to `execute` a program.